### PR TITLE
DGE test cases

### PR DESCRIPTION
**Pull request by amit-samal**
Added test cases for dge_lmem and dge_pseudobulk.

### CODE REVIEW

File: scalr/utils/data_utils.py [hunk 1]
```diff
@@ -82,3 +83,62 @@ def generate_dummy_anndata(n_samples, n_features, target_name='celltype'):
...
+    n_obs = obs.shape[0]
+    obs.index = obs.index.astype(str)
+
+    # Random geneexpression matrix
+    X = csr_matrix(np.random.rand(n_obs, n_vars))
+
+    # Creating var
+    var = pd.DataFrame({
+        'gene_id': [f'gid_{i}' for i in range(1, n_vars + 1)],
+        'gene_name': [f'G{i}' for i in range(1, n_vars + 1)]
```
Comment (mayur-iipl) [inline scalr/utils/data_utils.py:137]:
```suggestion
        'gene_name': [f'Gene_{i}' for i in range(1, n_vars + 1)]
```

**Review by mayur-iipl — COMMENTED** [review]
Some comments.

File: scalr/analysis/test_dge_lmem.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    dirpath = './tmp'
+    dge_lm = dge_lmem.DgeLMEM(fixed_effect_column='disease',
+                              fixed_effect_factors=['disease_x', 'normal'],
+                              group='donor_id',
+                              celltype_column='cell_type',
+                              cell_subsets=['B_cell', 'T_cell'])
+
+    dge_lm.generate_analysis(adata, dirpath)
+
+    lmem_dirpath = dirpath + '/lmem_dge_result'
```
[the remark this answers is not in the document]
Comment (amit-samal) [reply scalr/analysis/test_dge_lmem.py]: Included path.join

File: scalr/analysis/test_dge_lmem.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    result_files = os.listdir(lmem_dirpath)
+    csv_files = [file for file in result_files if file.endswith('.csv')]
+    svg_files = [file for file in result_files if file.endswith('.svg')]
```
[the remark this answers is not in the document]
Comment (amit-samal) [reply scalr/analysis/test_dge_lmem.py]: Combined these two loops into a single for loop.

File: scalr/utils/data_utils.py (hunk 1 above)
Comment (amit-samal) [reply scalr/utils/data_utils.py]: Changed from 'G_i' to 'gene_i' for gene names.

**Review by mayur-iipl — APPROVED** [review]

File: scalr/analysis/dge_pseudobulk.py [hunk 4]
```diff
@@ -59,7 +59,9 @@ def __init__(self,
         self.design_factor = design_factor
         # Hacky fix since pydeseq2 does not like `_` in column_names
```
Comment (anand-infocusp) [inline scalr/analysis/dge_pseudobulk.py:60]:
```suggestion
        # Hacky fix since pydeseq2 does not support `_` in column names.
```

Comment (amit-samal) [reply scalr/analysis/dge_pseudobulk.py:60]: Changed from 'like' to support'

**Review by mayur-iipl — APPROVED** [review]

**Review by mayur-iipl — APPROVED** [review]

**Review by anand-infocusp — APPROVED** [review]
lgtm

**Comment by saurabhinfocusp** [comment]
LGTM merging into main

**Pull request merged**